br: flaky test TestGCServiceSafePoint

## Bug Report

```
[2024/12/30 19:41:09.956 +00:00] [WARN] [advancer.go:478] ["failed to update global checkpoint: stale"] [old=5] [new=0]
[2024/12/30 19:41:09.956 +00:00] [WARN] [advancer.go:478] ["failed to update global checkpoint: stale"] [old=5] [new=4]
[2024/12/30 19:41:09.956 +00:00] [INFO] [advancer.go:620] ["updated log backup GC safe point."] [checkpoint=4] [target=4]
    advancer_test.go:210: 
        	Error Trace:	br/pkg/streamhelper/advancer_test.go:210
        	Error:      	Not equal: 
        	            	expected: 0x4
        	            	actual  : 0x3
        	Test:       	TestGCServiceSafePoint
--- FAIL: TestGCServiceSafePoint (0.00s)
FAIL
coverage: 1.4% of statements
--
Coverage runner: Not collecting coverage for failed test.
The following commands failed with status 1
```
easy to reproduce locally

--- a/br/pkg/streamhelper/advancer.go
+++ b/br/pkg/streamhelper/advancer.go
@@ -127,6 +127,9 @@ func newCheckpointWithSpan(s spans.Valued) *checkpoint {
 }
 
 func (c *checkpoint) safeTS() uint64 {
+	if c.TS == 0 {
+		return 0
+	}
 	return c.TS - 1
 }
 
@@ -429,7 +432,7 @@ func (c *CheckpointAdvancer) onTaskEvent(ctx context.Context, e TaskEvent) error
 		}
 		log.Info("get global checkpoint", zap.Uint64("checkpoint", globalCheckpointTs))
 		c.lastCheckpoint = newCheckpointWithTS(globalCheckpointTs)
-		p, err := c.env.BlockGCUntil(ctx, globalCheckpointTs-1)
+		p, err := c.env.BlockGCUntil(ctx, c.lastCheckpoint.safeTS())
 		if err != nil {
 			log.Warn("failed to upload service GC safepoint, skipping.", logutil.ShortError(err))
 		}

--- a/br/pkg/streamhelper/basic_lib_for_test.go
+++ b/br/pkg/streamhelper/basic_lib_for_test.go
@@ -677,7 +677,7 @@ func newTestEnv(c *fakeCluster, t *testing.T) *testEnv {
 		Name: "whole",
 		Info: &backup.StreamBackupTaskInfo{
 			Name:    "whole",
-			StartTs: 1,
+			StartTs: 0,
 		},
 		Ranges: rngs,
 	}
@@ -782,7 +782,7 @@ func (t *testEnv) putTask() {
 		Name: "whole",
 		Info: &backup.StreamBackupTaskInfo{
 			Name:    "whole",
-			StartTs: 5,
+			StartTs: 0,
 		},
 		Ranges: rngs,
 	}

--- a/br/pkg/streamhelper/basic_lib_for_test.go
+++ b/br/pkg/streamhelper/basic_lib_for_test.go
@@ -677,7 +677,7 @@ func newTestEnv(c *fakeCluster, t *testing.T) *testEnv {
 		Name: "whole",
 		Info: &backup.StreamBackupTaskInfo{
 			Name:    "whole",
-			StartTs: 5,
+			StartTs: 1,
 		},
 		Ranges: rngs,
 	}

